packages/discord.js/src/structures/Embed.js

```diff
@@ -211,9 +211,30 @@ class Embed {
    */
   equals(other) {
     if (other instanceof Embed) {
-      return isEqual(other.data, this.data);
+      return isEqual(this.data, other.data);
     }
-    return isEqual(other, this.data);
+
+    return (
+      this.color === (other.color ?? null) &&
+      this.description === (other.description ?? null) &&
+      this.title === (other.title ?? null) &&
+      this.url === (other.url ?? null) &&
+      (this.timestamp ? new Date(this.timestamp).getTime() : null) ===
+        (other.timestamp ? new Date(other.timestamp).getTime() : null) &&
+      this.image?.url === other.image?.url &&
+      this.thumbnail?.url === other.thumbnail?.url &&
+      this.video?.url === other.video?.url &&
+      isEqual(this.fields, other.fields?.map(field => ({ ...field, inline: field.inline ?? false })) ?? []) &&
+      isEqual(this.provider, other.provider ?? null) &&
+      isEqual(
+        { icon_url: this.author?.iconURL, name: this.author?.name, url: this.author?.url },
+        { icon_url: other.author?.icon_url, name: other.author?.name, url: other.author?.url },
+      ) &&
+      isEqual(
+        { icon_url: this.footer?.iconURL, text: this.footer?.text },
+        { icon_url: other.footer?.icon_url, text: other.footer?.text },
+      )
```

Comment:
Missing proxy URL comparisons. Is there a point in using `isEqual()` on 2 manually constructed objects here instead of direct comparisons?